app/views/users/new.html.haml

```diff
@@ -2,13 +2,16 @@
   .row
     .col-sm-10.col-sm-offset-1
       = bootstrap_form_for @user, layout: :horizontal, url: register_path do |f|
+        - if @invitation
+          = hidden_field_tag :invitation_token, @invitation.token
+
         %header
           %h1 Register
         %fieldset
           = f.email_field :email, control_col: 'col-sm-6', label: 'Email Address'
```

Comment:
I agree. But we want students to see the good/bad side of using such gem.

Good side is - when you know it already, it may power you up.

Bad side is - you have to get to know it. :D